src/pages/home/report/ReportActionsView.js

```diff
@@ -399,11 +399,15 @@ class ReportActionsView extends React.Component {
      * @param {Boolean} [shouldResetLocalCount=false] Whether count should increment or reset
      */
     updateLocalUnreadActionCount(shouldResetLocalCount = false) {
-        this.setState(prevState => ({
-            localUnreadActionCount: shouldResetLocalCount
+        this.setState((prevState) => {
+            const localUnreadActionCount = shouldResetLocalCount
```

Comment:
I think it would make more sense to set this variable and call `this.updateUnreadIndicatorPosition(localUnreadActionCount)` function first. Then call the `setState` function. Those two things aren't really a part of setting the state, so it seems that we could do them before as part of the `updateLocalUnreadActionCount` before calling `setState`.

Reply:
@puneetlath 
Something like this right?
```js
updateLocalUnreadActionCount(shouldResetLocalCount = false) {
        const localUnreadActionCount = shouldResetLocalCount
            ? this.props.report.unreadActionCount
            // eslint-disable-next-line react/no-access-state-in-setstate
            : this.state.localUnreadActionCount + this.props.report.unreadActionCount;
        this.updateUnreadIndicatorPosition(localUnreadActionCount);
        this.setState({localUnreadActionCount});
    }
```

Reply:
Yes, exactly!

Reply:
Will you need this though? 

`// eslint-disable-next-line react/no-access-state-in-setstate`

Reply:
> Will you need this though?
> 
> `// eslint-disable-next-line react/no-access-state-in-setstate`

Yes, it shows a warning not to access to state directly.

Reply:
@puneetlath 
<img width="889" alt="Screenshot 2021-10-05 at 7 09 58 PM" src="https://user-images.githubusercontent.com/85645967/136034466-e07f38c0-bf0c-4fa4-860a-3e154c9f6125.png">

It recommends using setState callback as you can see.

Reply:
Can I proceed with the changes mentioned here ->https://github.com/Expensify/App/pull/5534#discussion_r721913641

Reply:
Ok! I read more about it and it seems the way you were originally doing it is the preferred way, so we'll leave it as you originally had it. 

Thanks!